### ISSUE TITLE
How to customize react-datepicker-wrapper class

### ISSUE THREAD

**Issue by chupee120** [Hacker0x01/react-datepicker#2099]
Hello,

I'm not able to set width of datePicker to 100%

import "./customDatePickerWidth.css";

<div className="customDatePickerWidth">
   <DatePicker dateFormat="dd/MM/yyyy" />
</div>

.customDatePickerWidth, 
.customDatePickerWidth > div.react-datepicker-wrapper, 
.customDatePickerWidth > div > div.react-datepicker__input-container
.customDatePickerWidth > div > div.react-datepicker__input-container input {
   width: 100%;
}

also try this without success:

.react-datepicker__input-container {
    display: block;
}

.react-datepicker-wrapper {
    width: 100%;
}

**Comment by cordero67** [comment]
I'm having the same problem. Bumping this issue.

**Comment by danielmdebrito** [comment]
The solution can be found here: https://stackoverflow.com/questions/55794770/how-can-i-style-react-datepicker

**Comment by Cordazar** [comment]
But this should be a simple option or className set from the component, instead of adding global css rules to override it.

**Comment by BrandonMathis** [comment]
I agree with @Cordazar. Overriding styles with CSS would not be ideal because my changes would impact every instance of the datepicker in my application.

**Comment by wude935** [comment]
I'm having the same issue. The react-datepicker-class should be exposed via props so that styling is possible without editing all global instances, IMO.

**Comment by jaypeetancero** [comment]
I hope this concern would be resolve I don't want to create global style just for that.

**Comment by sonny89** [comment]
Try with `wrapperClassName` property:

```<DatePicker wrapperClassName="datePicker" dateFormat="dd/MM/yyyy" />```

```
.datePicker {
  width: 100%;
}
```

**Comment by rosghub** [comment]
@sonny89 confirming this works great

**Comment by isaacminogue** [comment]
Whilst the above styles the input itself, if you want to style the calendar component, I've had success with `<DatePicker calendarClassName={/* conditions */}></DatePicker>` with conditional statements to show different classes based on values in the parent component

**Comment by dexkode4** [comment]
awesome , very helpful

**Comment by marwajomaa** [comment]
> wrapperClassName="datePicker"

how to use this with next.js it's not working?!!

**Comment by undefineq** [comment]
> > wrapperClassName="datePicker"
> 
> how to use this with next.js it's not working?!!

import styles from "./styles.module.scss";

wrapperClassName={styles.datePicker}

**Comment by marwajomaa** [comment]
Yes, I did that but not working!!!

**Comment by No-you-are-not** [comment]
i have the same issue, when im trying to style the calendar component and not input. Nothing works for me, not calendarClassName, not wrapperClassName, not even putting datepicker inside of another div. The only thing that i was able to change styles is changing them by tags like this:
.wrapper div{
  border: none;
  background-color: var(--base-white);
}
but that is just too global obviously

**Comment by ismatim** [comment]
> > > wrapperClassName="datePicker"
> > 
> > 
> > how to use this with next.js it's not working?!!
> 
> import styles from "./styles.module.scss"; wrapperClassName={styles.datePicker}



> > wrapperClassName="datePicker"
> 
> how to use this with next.js it's not working?!!

There is no straight way to make it work with cssmodules & nextjs because it is needed to set each class with each element (inside datapicker).

**Comment by Ofer-Gal** [comment]
The 100% of the datepicker is not the same of other control in my page
![image](https://user-images.githubusercontent.com/5054931/150998482-2b77f9d9-ba16-4e08-a173-96e981d8767d.png)

I tried all sorts of ways. The closest to other controls 100% is 99% 😥

**Comment by trunglt-2769** [comment]
> The 100% of the datepicker is not the same of other control in my page ![image](https://user-images.githubusercontent.com/5054931/150998482-2b77f9d9-ba16-4e08-a173-96e981d8767d.png)
> 
> I tried all sorts of ways. The closest to other controls 100% is 99% 😥

Try adding props `box-sizing: border-box;` to input element of React DatePicker

**Comment by afif14** [comment]
> There is no straight way to make it work with cssmodules & nextjs because it is needed to set each class with each element (inside datapicker). (as far I could see)

and how to make it work in next js, i have the same issue

**Comment by Yanger-Rai** [comment]
### For Next.js
If you are using shadcn UI. Go into calendar component and remove "flex" from row and head_row. Works perfectly. Refer the screenshot below. Haven't tried directly in day picker, maybe someone do it.

<img width="493" alt="Screenshot 2023-08-01 at 1 47 40 AM" src="https://github.com/Hacker0x01/react-datepicker/assets/47139792/adbfc648-fe28-426f-afbf-acad74663bc9">

**Comment by rever96** [comment]
> Try with `wrapperClassName` property:
> 
> `<DatePicker wrapperClassName="datePicker" dateFormat="dd/MM/yyyy" />`
> 

This issue should be closed with this reply

**Comment by oseji** [comment]
> ### For Next.js
> If you are using shadcn UI. Go into calendar component and remove "flex" from row and head_row. Works perfectly. Refer the screenshot below. Haven't tried directly in day picker, maybe someone do it.
> 
> <img alt="Screenshot 2023-08-01 at 1 47 40 AM" width="493" src="https://private-user-images.githubusercontent.com/47139792/257338339-adbfc648-fe28-426f-afbf-acad74663bc9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NzczMjksIm5iZiI6MTcxNzc3NzAyOSwicGF0aCI6Ii80NzEzOTc5Mi8yNTczMzgzMzktYWRiZmM2NDgtZmUyOC00MjZmLWFmYmYtYWNhZDc0NjYzYmM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDE2MTcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5NjZjNjM1NDdmZWM4ODNiZjRhMzdlYmYxMDZkYjYzY2I3MmRjNjkzNmM0YjkxZmVmYzZjNzUzNThhYTAyNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8DvoggfZrBspXJswP0Ct59TDwtuoiQIE7CF1qQlnnMQ">

how can i change the width of the calendar container

**Comment by naufalandya** [comment]
i did it with tailwind and it works

                    `<DatePicker
                      selected={deadlineTime}
                      onChange={(date) => setDeadlineTime(date)}
                      showTimeSelect
                      dateFormat="Pp"
                      customInput={
                        <Input
                          width={'100%'}
                          type="text"
                          value={deadlineTime ? deadlineTime.toLocaleString() : ''}
                          readOnly
                          className="bg-gray-800 text-white border-none p-2 rounded"
                        />
                      }
                      wrapperClassName="w-full"
                      popperClassName="w-full"
                      className="w-full bg-gray-800 text-white border-none p-2 rounded"
                      calendarClassName="bg-gray-800"
                      dayClassName={() => "text-gray-400"}
                      monthClassName={() => "text-white"}
                      timeClassName={() => "text-gray-400"}
                    />`

**Comment by dalvarez** [comment]
Super not-ideal way around it all, for those on the Tailwind boat:
If you want a way to style the elements that come with the library but find that many of the props for styles aren't working, you can target specific classes with arbitrary variants like this
`[&_[class*='day-names']]:bg-white`

This will find the element with the class attribute whose substring contains "day-names." Personally, I've added these sort of rules to my customContainer div.

**Comment by Marlstergo** [comment]
Okay so here is what worked for me
` <DatePicker
calendarClassName="react-datepicker-popper" />
`  then you can directly overwrite the styles `.react-datepicker__month-container {
  float: unset !important;
  width: 100% !important;
  /* other styles */
}`

**Comment by anyderre** [comment]
Rewrite the class with name with your desired css in a custom css file, like this. It should work. I wanted it like this `.react-date-picker__wrapper {
  display: flex;
  flex-grow: 1;
  flex-shrink: 0;
  border: 0 !important; // The change I wanted to make
}` 

It works for me maybe if you give it some try it could work for you too.

**Comment by pagespeedattacking** [comment]
> Whilst the above styles the input itself, if you want to style the calendar component, I've had success with `<DatePicker calendarClassName={/* conditions */}></DatePicker>` with conditional statements to show different classes based on values in the parent component





https://reactdev.space/questions/can-the-month-dropdown-in-react-datepicker-be-modified-to-display-numbers-instead-of-names

**Comment by gileswebberley** [comment]
I had a bit of heartbreak when I went to look at my work from through the night, all of the custom styling that I had spent ages applying no longer had an effect. I'm using styled component's createGlobalStyle which is probably where the problem arises but I have thankfully just tried adding !important to everything and it seems to be coming back to where I'd left it. I'm not very experienced so I'm sure there is a better solution but it's working for me and I hope it helps someone else out there :)
